tests/settings_tests/test_deprecation.py

```diff
@@ -0,0 +1,43 @@
+import sys
+from types import ModuleType
+
+from django.conf import Settings, settings
+from django.test import SimpleTestCase, ignore_warnings
+from django.utils.deprecation import RemovedInDjango31Warning
+
+
+class DeprecationTests(SimpleTestCase):
+    msg = (
+        'The FILE_CHARSET setting is deprecated. Starting with Django 3.1, all '
+        'files read from disk must be UTF-8 encoded.'
+    )
+
+    def test_override_settings_warning(self):
+        with self.assertRaisesMessage(RemovedInDjango31Warning, self.msg):
+            with self.settings(FILE_CHARSET='latin1'):
+                pass
+
+    def test_settings_init_warning(self):
+        settings_module = ModuleType('fake_settings_module')
+        settings_module.FILE_CHARSET = 'latin1'
+        settings_module.SECRET_KEY = 'abc'
+        sys.modules['fake_settings_module'] = settings_module
+        try:
+            with self.assertRaisesMessage(RemovedInDjango31Warning, self.msg):
+                Settings('fake_settings_module')
+        finally:
+            del sys.modules['fake_settings_module']
+
+    def test_access_warning(self):
+        with self.assertRaisesMessage(RemovedInDjango31Warning, self.msg):
+            settings.FILE_CHARSET
+        # Works a second time.
```

Comment:
Curious about the reason why it may not work a second time (and it's unclear to me that it needs to work a second time, maybe one warning is enough?).